Source/AwsCommonRuntimeKit/http/HttpClientConnection.swift

```diff
@@ -23,102 +26,24 @@ public class HttpClientConnection {
     }
 
     /// Close the http connection
-    public func close() {
-        manager.releaseConnection(connection: self)
-    }
+// TODO: do we need a explicit close function or deinit is enough?
+//    public func close() throws
+//        try manager.releaseConnection(connection: self)
+//        manager = nil
+//    }
```

Comment:
Yeah, it's common enough that people want to kill a connection because something is going wrong, but getting its refcount down to zero is a complicated process.

but ughghhgh I'm not sure what's going on with this design where "close" really means "give it back to the connection manager for re-use".

In C, the HttpClientConnection is ignorant of the HttpClientConnectionManager, but here in Swift we've tied their implementation together.

I really need to look this through 🤦‍♀️

Reply:
Thanks, resolved.